src/rust/engine/src/externs/interface.rs

```diff
@@ -1392,28 +1392,23 @@ fn ensure_remote_has_recursive(
 fn single_file_digests_to_bytes<'py>(
   py: Python<'py>,
   py_scheduler: &PyScheduler,
-  py_file_digests: &PyList,
+  py_file_digests: Vec<PyFileDigest>,
 ) -> PyO3Result<&'py PyList> {
   let core = &py_scheduler.0.core;
   core.executor.enter(|| {
-    let digests: Vec<Digest> = py_file_digests
-      .iter()
-      .map(|item| crate::nodes::lift_file_digest(&core.types, item))
-      .collect::<Result<Vec<Digest>, _>>()
-      .map_err(PyException::new_err)?;
-
-    let digest_futures = digests.into_iter().map(|digest| {
+    let digest_futures = py_file_digests.into_iter().map(|py_file_digest| {
       let store = core.store();
       async move {
         store
-          .load_file_bytes_with(digest, |bytes| {
+          .load_file_bytes_with(py_file_digest.0, |bytes| {
             let gil = Python::acquire_gil();
             let py = gil.python();
             externs::store_bytes(py, bytes)
           })
```

Comment:
I opened https://github.com/PyO3/pyo3/issues/1992 to propose teaching PyO3 about `bytes::Bytes` so that we can avoid using the GIL inside this future.